docs/docs/jsx-in-depth.md

```diff
@@ -0,0 +1,312 @@
+---
+id: jsx-in-depth
+title: JSX In Depth
+permalink: docs/jsx-in-depth.html
+---
+
+Fundamentally, JSX just provides syntactic sugar for the `React.createElement(component, props, children)` function. The JSX code:
+
+```js
+<MyComponent prop1={'one'} prop2={'two'}>stuff</MyComponent>
```

Comment:
That, and let's use double quotes in JSX too since most popular styleguides do that. So it'll be `prop1="one"`.

I would also like us to use more real-world over contrived examples. Like Button or Avatar instead of MyComponent or Foo.